src/_macosx.m

```diff
@@ -1526,16 +1537,17 @@ - (void)otherMouseDragged:(NSEvent *)event { [self mouseDragged: event]; }
 
 - (void)setRubberband:(NSRect)rect
 {
-    if (!NSIsEmptyRect(rubberband)) { [self setNeedsDisplayInRect: rubberband]; }
     rubberband = rect;
-    [self setNeedsDisplayInRect: rubberband];
+    // We can't just set DisplayInRect because the outer paths need to be removed
+    // when shrinking the zoom-box inwards
+    [self setNeedsDisplay: YES];
```

Comment:
You could take the union of the existing `rubberband` with the new `rect` (assuming ObjC has that API)?

Reply:
Yep, good call, updated with that now.